Sources/TuistKit/Commands/CommandRegistry.swift

```diff
@@ -104,14 +105,14 @@ public final class CommandRegistry {
         return arguments[1]
     }
 
-    private func parse() throws -> ArgumentParser.Result? {
+    private func parse() throws -> (ArgumentParser.Result, ArgumentParser)? {
         let arguments = Array(processArguments().dropFirst())
         guard let argumentName = arguments.first else { return nil }
         let subparser = try parser.parse([argumentName]).subparser(parser)
         if let command = commands.first(where: { type(of: $0).command == subparser }) {
             return try command.parse(with: parser, arguments: arguments)
         }
-        return try parser.parse(arguments)
+        return (try parser.parse(arguments), parser)
```

Comment:
parser is an instance on `CommandRegistry`, I'm not sure what the benefit of returning it here is?

Reply:
`SigningCommand` returns a custom parser to make it easier to parse subcommands, I'll try to revisit it if we can make it work without creating a new one

Reply:
Unfortunately, I could not do it -> the reason is because [here](https://github.com/tuist/tuist/pull/1127/files#diff-7e3710f4470c93590881d6a91fdd047eR130) I need to trigger a command which is found by `subparser`. `subparser` can be only one word, thus I need a custom parser that is `tuist signing` and then I can successfully find the subcommand.

I am not happy with this extra logic, but we should start using the new Swift argument parser sooner than later which supports subcommands out of the box, so maybe perfecting the mechanism might be lost work, so I'd prefer keeping it as it is.